circuit/types/boolean/src/not.rs

```diff
@@ -38,6 +38,8 @@ impl<E: Environment> Not for &Boolean<E> {
             // Constant case.
             true => Boolean(E::one() - &self.0),
             // Public and private cases.
+            // Note: We must directly instantiate a public variable due to the way we represent a boolean in a linear combination.
+            // For more information, see `LinearCombination::is_boolean_type`.
             false => Boolean(Variable::Public(0, Rc::new(E::BaseField::one())) - &self.0),
```

Comment:
Can this not be `E::one()`?

Reply:
i.e.

```suggestion
            false => Boolean(E::one()),
```

Reply:
It produces an error saying that the boolean is not well formed.
```
Both LC::constant and LC::terms contain elements, which is a violation
thread 'not::tests::test_not_private' panicked at 'Boolean variable is not well-formed', []/snarkVM/circuit/environment/src/circuit.rs:246:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Related, what's the intuition for representing boolean values in this way?
https://github.com/AleoHQ/snarkVM/blob/0c20d9ae0d96b0d8704e85ce682551c43153b072/circuit/environment/src/helpers/linear_combination.rs#L91

Reply:
This representation is a compact way to represent boolean values, with efficient operations.